src/hooks/useTransaction.ts

```
@@ -1,139 +1,191 @@
+
+	// Advance queue or complete on single tx success
+	useEffect(() => {
+		if (!isSuccess || canBatch || currentCalls.length === 0 || !txHash) return;
+		if (txHash === handledReceiptHash.current) return;
+		handledReceiptHash.current = txHash;
+
+		const nextIndex = queueIndex + 1;
+
+		if (nextIndex < currentCalls.length) {
+			// Send the next transaction in the queue
+			setQueueIndex(nextIndex);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/hooks/useTransaction.ts:119:4
  117 | 		if (nextIndex < currentCalls.length) {
  118 | 			// Send the next transaction in the queue
> 119 | 			setQueueIndex(nextIndex);
      | 			^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  120 | 			const next = currentCalls[nextIndex];
  121 | 			mutateTransaction({ to: next.to, data: next.data, value: next.value });
  122 | 		} else if (!hasCalledOnSuccess.current) {

(react-hooks/set-state-in-effect)